tests/tripal_chado/fields/ChadoFieldGetValuesListTest.php

```diff
@@ -0,0 +1,442 @@
+  // Uncomment to auto start and rollback db transactions per test method.
+  use DBTransaction;
+
+  // Stores a list of field instances to be tested including their storage method and instance info.
+  private $field_list = NULL;
+
+  /**
+   * Test getValueList for fields based on columns in the base table.
+   *
+   * @dataProvider getBaseFields
+   *
+   * @group fields
+   * @group getValueList
+   */
+  public function testBaseTableColumns($field_name, $bundle_name, $info) {
+    include_once(drupal_get_path('tripal_chado', 'module') . '/includes/TripalFields/ChadoField.inc');
+    $faker = \Faker\Factory::create();
+
+    // Construct the Field instance we want the values for.
+    // Specifying "ChadoField" here ensures we are only testing our
+    // implementation of getValueList() and not the custom version for any
+    // given field.
+    // YOU SHOULD TEST CUSTOM FIELD IMPLEMENTATIONS SEPARATELY.
+    $instance = new \ChadoField($info['field_info'], $info['instance_info']);
+
+    // Retrieve the values.
+    // $values will be an array containing the distinct set of values for this field instance.
+    $values = $instance->getValueList(array('limit' => 5));
+
+    // Ensure we have values returned!
+    $this->assertTrue(
+      is_array($values),
+      t(
+        'No values returned for @field_name (bundle: @bundle_name, bundle base table: @bundle_base_table, chado table: @chado_table, chado column: @chado_column).',
+        array(
+          '@field_name' => $field_name,
+          '@bundle_name' => $bundle_name,
+          '@bundle_base_table' => $info['bundle_base_table'],
+          '@chado_table' => $info['instance_info']['settings']['chado_table'],
+          '@chado_column' => $info['instance_info']['settings']['chado_column'],
+        )
+      )
+    );
+
+    // Ensure there are no more then 5 as specified in the limit above.
+    $this->assertLessThanOrEqual(5, sizeof($values),
+      t('Returned too many results for @field_name.', array('@field_name' => $field_name)));
+
+    // Ensure a known value is in the list.
+    // Note: The following generates fake data with a fixed value for the column this
+    // field is based on. This allows us to check that fixed value is one of those
+    // returned by ChadoField->getValueList().
+    $fake_value = $this->generateFakeData($info['bundle_base_table'], $info['base_schema'], $info['instance_info']['settings']['chado_column']);
+    if ($fake_value !== FALSE) {
+
+      // Re-generate the values...
+      $values = $instance->getValueList(array('limit' => 200));
+
+      // And ensure our fake value is in the returned list.
+      // We can only check this if all the results are returned.
+      // As such, we set the limit quite high above and if we have
+      // less then the limit, we will go ahead with the test.
+      // @note: this tests all fields on TravisCI since there is no pre-existing data.
+      if (sizeof($values) < 200) {
```

Comment:
so this test wont execute if i have over 200 records.  Is that as intended?

Reply:
Looks like this should be an assertion `$this->assertLessThanOrEqual(200, sizeof($values)`?

Reply:
It's intended for the nested assertions to not run if there are more then 200 records. 

This is due to the fact that I can't guarantee my fixed fake values will be in the subset returned. We need this because I was unable to create fake bundles and as such there may be existing data in the database. We don't want the test to fail since it doesn't actually indicate the method is not working, it's simply that we can't test as well as I would like at this point.